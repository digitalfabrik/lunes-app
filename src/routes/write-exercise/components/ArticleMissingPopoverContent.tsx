import React, { ReactElement } from 'react'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
import styled from 'styled-components/native'

import { InfoIcon } from '../../../../assets/images'
import labels from '../../../constants/labels.json'

const StyledContainer = styled.View`
  flex-direction: row;
  align-items: center;
  background-color: ${prop => prop.theme.colors.lunesBlack};
  width: ${wp('80%')}px;
  height: 80px;
  padding: 9px 8px 9px 8px;
  border-radius: 2px;
`
const StyledMessage = styled.Text`
  color: ${prop => prop.theme.colors.lunesWhite};
  font-size: ${props => props.theme.fonts.smallFontSize};
  font-weight: ${props => props.theme.fonts.lightFontWeight};
  font-family: ${props => props.theme.fonts.contentFontRegular};
  width: ${wp('60%')}px;
  margin-left: 8px;
`

const ArticleMissingPopoverContent = (): ReactElement => (
  <StyledContainer>
    <InfoIcon width={30} height={30} />
    <StyledMessage>{labels.exercises.write.feedback.articleMissing}</StyledMessage>
  </StyledContainer>
)

export default ArticleMissingPopoverContent
