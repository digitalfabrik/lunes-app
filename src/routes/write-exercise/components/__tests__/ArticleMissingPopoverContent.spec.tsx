import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import 'react-native'

import ArticleMissingPopoverContent from '../ArticleMissingPopoverContent'

describe('components', () => {
  describe('ArticleMissingPopoverContent', () => {
    it('renders correctly across screens', () => {
      const component = shallow(<ArticleMissingPopoverContent />)
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
