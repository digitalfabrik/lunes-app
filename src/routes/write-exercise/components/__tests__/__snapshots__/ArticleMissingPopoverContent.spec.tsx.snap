// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ArticleMissingPopoverContent renders correctly across screens 1`] = `
<Styled(View)>
  <SvgrURL
    height={30}
    width={30}
  />
  <Styled(Text)>
    Ups, deine Eingabe ist nicht vollständig. Füge einen Artikel ein.
  </Styled(Text)>
</Styled(View)>
`;
